R/prepare_matches_mapper.R

```diff
@@ -12,10 +12,11 @@ prepare_matches_mapper <- function(mapper, activities) {
   mapper |>
     left_join(activities, by = "activity_uuid_product_uuid") |>
     summarise(across(everything(), ~ paste0(na.omit(unique(.)), collapse = "; ")), .by = c("country", "main_activity", "clustered")) |>
-    ungroup() |>
+    mutate(across(everything(), ~ifelse(. == "", NA_character_, .))) |>
```

Comment:
This line seems correct but could improve a bit:

* `everything()` is misleading since we only operate on strings. Thus it's more precise to use `across(where(is.character), ...`.
* `. == ""` is clear but `nzchar()` is more focused and likely faster.
* I think the tidyverse team seems to be moving away from creating lambdas with `~` in favor or either `function(x)`, `\(x)` (since R 4.0), or just creating a named function.
``` r
library(dplyr, warn.conflicts = FALSE)

data <- tibble::tibble(x = c(TRUE, FALSE), y = c("", "a"))
data
#> # A tibble: 2 × 2
#>   x     y    
#>   <lgl> <chr>
#> 1 TRUE  ""   
#> 2 FALSE "a"

# `?nzchar`: nzchar is a fast way to find out if elements of a character vector are non-empty strings.
data |> mutate(across(where(is.character), \(x) ifelse(!nzchar(x), NA_character_, x)))
#> # A tibble: 2 × 2
#>   x     y    
#>   <lgl> <chr>
#> 1 TRUE  <NA> 
#> 2 FALSE a

replace_empty_with_na <- function(x) ifelse(!nzchar(x), NA_character_, x)
data |> mutate(across(where(is.character), replace_empty_with_na))
#> # A tibble: 2 × 2
#>   x     y    
#>   <lgl> <chr>
#> 1 TRUE  <NA> 
#> 2 FALSE a
```

<sup>Created on 2023-10-13 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
@kalashsinghal we have a logical column `multi_match` which is not the character column, hence I have added `everything()` in the new commit in all the commands. Please let me know if you would like to see this change!

I have applied all the updates you recommended. Thanks for taking out so much time on this!